From 22cd9571827e9a2dbb4c93b271e04ef3751b2b5e Mon Sep 17 00:00:00 2001
From: Shen Zhenyi <szy@rock-chips.com>
Date: Fri, 30 Oct 2015 13:59:55 +0800
Subject: [PATCH] hdmi: cec: add HDMI_IOCTL_CECWAKESTATE

android HDMI CEC HAL need to know cec state when resume,
so we add the HDMI_IOCTL_CECWAKESTATE.

Change-Id: Iff942c6085d10891200cd873cc2ba53e9a2d6cad
Signed-off-by: Shen Zhenyi <szy@rock-chips.com>
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-cec.c | 3 +++
 drivers/video/rockchip/hdmi/rockchip-hdmi-cec.h | 4 +++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.c
index 68c7bf115152..f7e25f844245 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.c
@@ -226,6 +226,9 @@ static long cec_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		break;
 	case HDMI_IOCTL_CECCLEARLA:
 		break;
+	case HDMI_IOCTL_CECWAKESTATE:
+		ret = copy_to_user(argp, &(cec_dev->hdmi->sleep), sizeof(int));
+		break;
 
 	default:
 		break;
diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.h b/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.h
index 86d94c45b193..cedb02d2d5f8 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.h
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-cec.h
@@ -184,7 +184,9 @@ struct cec_device {
 #define HDMI_IOCTL_CECREAD   _IOR(HDMI_CEC_MAGIC, 4, struct cec_framedata)
 #define HDMI_IOCTL_CECSETLA  _IOW(HDMI_CEC_MAGIC, 5, int)
 #define HDMI_IOCTL_CECCLEARLA  _IOW(HDMI_CEC_MAGIC, 6, int)
+#define HDMI_IOCTL_CECWAKESTATE  _IOR(HDMI_CEC_MAGIC, 7, int)
 /*for HAL ioctl end*/
+
 int rockchip_hdmi_cec_init(struct hdmi *hdmi,
 			   int (*sendframe)(struct hdmi *,
 					    struct cec_framedata *),
@@ -193,4 +195,4 @@ int rockchip_hdmi_cec_init(struct hdmi *hdmi,
 			   void (*setceclogicaddr)(struct hdmi *, int));
 void rockchip_hdmi_cec_set_pa(int devpa);
 void rockchip_hdmi_cec_submit_work(int event, int delay, void *data);
-#endif /* __HDMI_CEC_H__ */
\ No newline at end of file
+#endif /* __HDMI_CEC_H__ */
-- 
2.35.3

